docs/src/content/docs/guides/mcp.mdx

```diff
@@ -97,6 +97,27 @@ For **Streamable HTTP** and **Stdio** servers, each time an `Agent` runs it may
 
 Only enable this if you're confident the tool list won't change. To invalidate the cache later, call `invalidateToolsCache()` on the server instance.
 
+### Tool filtering
+
+You can restrict which tools are exposed from each server. Pass either a static filter
+using `createMCPToolStaticFilter` or a custom function:
+
+```ts
```

Comment:
We consistently use compiled code snippets in docs. can you do this?

- Add `examples/docs/mcp/tool-filter.ts`, which compiles and works
- Add `import stdioExample from '../../../../../examples/docs/mcp/tool-filter.ts?raw';` and use it here

If you're unsure or don't have time for this, please just remove this change.